specutils/spectra/spectrum_collection.py

```diff
@@ -91,16 +91,23 @@ def __init__(self, flux, spectral_axis=None, wcs=None, uncertainty=None,
         self._meta = meta
 
     def __getitem__(self, key):
-        if isinstance(key, int) and self.ndim > 2:
+        flux = self.flux[key]
+        if flux.ndim != 1:
             raise ValueError("Currently only 1D data structures may be "
                              "returned from slice operations.")
-
-        return Spectrum1D(flux=self.flux[key],
-                          spectral_axis=self.spectral_axis[key],
-                          uncertainty=self.uncertainty[key],
-                          wcs=self.wcs[key],
-                          mask=self.mask[key],
-                          meta=self.meta[key])
+        spectral_axis = self.spectral_axis[key]
+        uncertainty = None if self.uncertainty is None else self.uncertainty[key]
+        wcs = None if self.wcs is None else self.wcs[key]
+        mask = None if self.mask is None else self.mask[key]
+        try:
+            meta = self.meta[key]
+        except:
```

Comment:
In general we shouldn't allow open except cases -- perhaps except on a `KeyError` here?

Reply:
I see your point - I was thinking to be open-ended on purpose because I realized that `meta = None` yields a different exception (`TypeError`)... But I guess it probably makes sense to just catch those *two*.